src/com/ableton/VirtualEnv.groovy

```diff
@@ -15,12 +15,59 @@ class VirtualEnv implements Serializable {
    * Destination directory for the virtualenv. This value is set during construction of
    * the object, and is under the system temporary directory.
    */
-  String destDir
+  String destDir = null
+  /**
+   * Series of commands needed to install and active a version of Python inside the
+   * current shell.
+   */
+  String envSetup = null
+
+  protected String activateSubDir = null
+
+  /**
+   * Create a virtualenv using a specific version of Python, installed via pyenv. pyenv
+   * should be installed on the node, but the actual setup of any required environment
+   * variables (e.g. PYENV_ROOT and PATH) will be done inside this function.
+   *
+   * @param script Script context.
+   *               <strong>Required value, may not be null!</strong>
+   * @param python Python version, as given by pyenv versions --list.
+   * @param pyenvRoot Path to the installation of pyenv.
+   * @return New instance of VirtualEnv object.
+   */
+  static VirtualEnv create(Object script, String python, String pyenvRoot) {
+    if (!script.isUnix()) {
+      script.error 'This method is not supported on Windows'
+    }
 
-  protected String activateSubDir
+    VirtualEnv venv = new VirtualEnv(script, python)
+    assert pyenvRoot
+
+    if (!script.fileExists(pyenvRoot)) {
+      script.error "pyenv root path '${pyenvRoot}' does not exist"
+    }
+
+    venv.envSetup = """
+      export PYENV_ROOT=${pyenvRoot}
+      export PATH=\$PYENV_ROOT/bin:\$PATH
+      eval "\$(pyenv init -)"
+    """
+
+    venv.script.sh("""
+      ${venv.envSetup}
+      pyenv install -s ${python}
+      pyenv shell ${python}
+      pip install virtualenv
+      virtualenv ${venv.destDir}
+    """)
+
+    venv.envSetup += ". ${venv.destDir}/${venv.activateSubDir}/activate"
```

Comment:
This is a good point—I'll have to let @nre-ableton comment here because he wrote this part 🙂

Reply:
@dgl-ableton I believe you are mistaken, but I like your paranoia. 😄  On Line 50 above (https://github.com/AbletonDevTools/python-pipeline-utils/pull/56/files#diff-2424724bbebf6d79ee5c8cdb6edc3e0dR50) we overwrite `venv.envSetup` by assignment. So this append operator should be safe.

Reply:
I see :D Maybe this is a silly question, but then why are we appending `venv.envSetup`?

Reply:
Because we must run the above `sh` call _without_ activating the venv. So what that script is doing is effectively:

1. Add environment variables needed for pyenv first
2. Add a call to `pyenv init -`
3. Use these values to ensure that the desired python version is installed
4. Install virtualenv itself
5. Create a virtualenv for the project

We can't add a call to `. ${venv.destDir}/${venv.activateSubDir}/activate` to `venv.envSetup` right away because the venv only gets created in steps 4-5. To avoid this chicken-and-egg problem, we just append the `activate` bit to `venv.envSetup` after we have created the venv itself.